src/app/Application/Middleware/IsSuccessMiddleware.cs

```diff
@@ -0,0 +1,170 @@
+    {
+        var paths = openApiDocument.Paths.Values;
+
+        foreach (var path in paths)
+        {
+            var operations = path.Operations.Values.FirstOrDefault();
+            if (operations is null)
+            {
+                continue;
+            }
+
+            var changesToMake = new List<KeyValuePair<string, OpenApiResponse>>();
+            var keysToRemove = new List<string>();
+
+            foreach (var response in operations.Responses)
+            {
+                var responseKey = response.GetResponseKey();
+
+                var content = response.Value.Content;
+                if (content.Count > 0)
+                {
+                    content.First().Value.Schema.Properties[IsSuccessField] = OpenApiSuccessSchema;
+                }
+                else
+                {
+                    content.Add(ContentType, new()
+                    {
+                        Schema = new()
+                        {
+                            Properties = new Dictionary<string, OpenApiSchema>
+                            {
+                                [IsSuccessField] = responseKey?.IsSuccessStatusKey() is true ? OpenApiSuccessSchema : OpenApiFailureSchema
+                            }
+                        }
+                    });
+                }
+
+                if (responseKey is NoContentStatusCode)
+                {
+                    response.Value.Description = "Success";
+
+                    changesToMake.Add(new(SuccessStatusCode.ToString(), response.Value));
+                    keysToRemove.Add(response.Key);
+                }
+            }
+
+            foreach (var change in changesToMake)
+            {
+                operations.Responses.Add(change.Key, change.Value);
+            }
+
+            foreach (var key in keysToRemove)
+            {
+                operations.Responses.Remove(key);
+            }
+        }
+
+        if (openApiDocument.Components.Schemas.TryGetValue(ProblemDetailsSchemaName, out var problemDetails))
+        {
+            problemDetails.Properties[IsSuccessField] = OpenApiFailureSchema;
+        }
+    }
+
+    private static async Task AddIsSuccessFieldInResponseBody(HttpContext context, RequestDelegate next)
+    {
+        var originalBodyStream = context.Response.Body;
+        using var newBodyStream = new MemoryStream();
+        context.Response.Body = newBodyStream;
+
+        await next.Invoke(context);
+
+        var isSuccess = context.Response.StatusCode.IsSuccessStatusKey();
+
+        context.Response.Body.Seek(0, SeekOrigin.Begin);
+        var responseBody = await new StreamReader(context.Response.Body).ReadToEndAsync(context.RequestAborted);
+        context.Response.Body.Seek(0, SeekOrigin.Begin);
+
+        if (string.IsNullOrEmpty(responseBody) is false)
+        {
+            var originalJson = JsonDocument.Parse(responseBody).RootElement;
```

Comment:
I think it would be nice if you modified the response body only if the response type is "application/json"

Reply:
Is also may be "application/problem+json" content type. So it'd be better to check if type contains "json" string